container/tomcat/src/main/java/org/jboss/modcluster/container/tomcat/ModClusterListener.java

```diff
@@ -255,181 +245,277 @@ public void setLoadMetricCapacity(String capacity) {
     }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#getProxyConfiguration()
-     */
     @Override
     public Map<InetSocketAddress, String> getProxyConfiguration() {
         return this.service.getProxyConfiguration();
     }
-    public String getProxyConfigurationString() {
-        String result = null;
-
-        Map<InetSocketAddress, String> map = this.service.getProxyConfiguration();
-        if (map.isEmpty())
-                return null;
-        Object results[] = map.values().toArray();
-        result = (String) results[0];
-        return result;
-    }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#getProxyInfo()
-     */
     @Override
     public Map<InetSocketAddress, String> getProxyInfo() {
         return this.service.getProxyInfo();
     }
-    public String getProxyInfoString() {
-        String result = null;
 
-        Map<InetSocketAddress, String> map = this.service.getProxyInfo();
-        if (map.isEmpty())
-                return null;
-        Object results[] = map.values().toArray();
-        result = (String) results[0];
-        return result;
-    }
-
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#refresh()
-     */
     @Override
     public void refresh() {
         this.service.refresh();
     }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#removeProxy(String, int)
-     */
     @Override
     public void removeProxy(String host, int port) {
         this.service.removeProxy(host, port);
     }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#reset()
-     */
     @Override
     public void reset() {
         this.service.reset();
     }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#stop(long, TimeUnit)
-     */
     @Override
     public boolean stop(long timeout, TimeUnit unit) {
         return this.service.stop(timeout, unit);
     }
-    public boolean stop(long timeout) {
-        return this.service.stop(timeout, TimeUnit.SECONDS);
-    }
 
-    /**
-     * {@inhericDoc}
-     *
-     * @see ModClusterServiceMBean#stopContext(String, String, long,
-     *      TimeUnit)
-     */
     @Override
     public boolean stopContext(String host, String path, long timeout, TimeUnit unit) {
         return this.service.stopContext(host, path, timeout, unit);
     }
 
+    // ---------------------- RHQ support methods introduced in https://bugzilla.redhat.com/show_bug.cgi?id=822250 ----------------------
+
+    // FIXME Why do these two following methods return only the first proxy configuration?
+    // FIXME Can we fix this without breaking RHQ?
+    public String getProxyConfigurationString() {
+        String result = null;
+
+        Map<InetSocketAddress, String> map = this.service.getProxyConfiguration();
+        if (map.isEmpty())
+            return null;
+        Object results[] = map.values().toArray();
```

Comment:
@rhusar I realize this was moved from elsewhere, but it's pretty inefficient to allocate an array, copy the whole collection into the array, just to obtain the 1st element. Better would be: map.values.stream().findFirst().orElse(null);